core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestRewriteCaseExpressionPredicate.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner.iterative.rule;
+
+import io.trino.sql.planner.iterative.rule.test.BaseRuleTest;
+import io.trino.sql.planner.iterative.rule.test.PlanBuilder;
+import io.trino.sql.tree.Expression;
+import org.testng.annotations.Test;
+
+import static io.trino.SystemSessionProperties.OPTIMIZE_CASE_EXPRESSION_PREDICATE;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.filter;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.values;
+
+public class TestRewriteCaseExpressionPredicate
+        extends BaseRuleTest
+{
+    @Test
+    public void testRewriterDoesNotFireOnPredicateWithoutCaseExpression()
+    {
+        assertRewriteDoesNotFire("col1 > 1");
+    }
+
+    @Test
+    public void testRewriterDoesNotFireOnPredicateWithoutComparisonFunction()
+    {
+        assertRewriteDoesNotFire("(case when col1=1 then 'case1' when col2=2 then 'case2' else 'default' end)");
+    }
+
+    @Test
+    public void testRewriterDoesNotFireOnPredicateWithFunctionCallOnComparisonValue()
+    {
+        assertRewriteDoesNotFire("(case when col1=1 then 'case1' when col2=2 then 'case2' else 'default' end) = upper('case1')");
+        assertRewriteDoesNotFire("(case when col1=1 then 10 when col2=2 then 20 else 30 end) = ceil(col1)");
```

Comment:
In this test, `col1` and `col2` should be the same.
Anyway, I didn't notice we had the condition you're testing here.

Reply:
Done.